kornia/feature/laf.py

```diff
@@ -115,30 +116,29 @@ def set_laf_orientation(LAF: torch.Tensor, angles_degrees: torch.Tensor) -> torc
     return laf_out
 
 
-def laf_from_center_scale_ori(xy: torch.Tensor, scale: torch.Tensor, ori: torch.Tensor) -> torch.Tensor:
+def laf_from_center_scale_ori(xy: torch.Tensor,
+                              scale: Optional[torch.Tensor] = None,
+                              ori: Optional[torch.Tensor] = None) -> torch.Tensor:
     """Return orientation of the LAFs, in radians. Useful to create kornia LAFs from OpenCV keypoints.
 
     Args:
         xy: tensor [BxNx2].
-        scale: tensor [BxNx1x1].
-        ori: tensor [BxNx1].
+        scale: tensor [BxNx1x1]. If not provided, scale = 1 is assumed
+        ori: tensor [BxNx1]. If not provided orientation = 0 is assumed
 
     Returns:
         tensor BxNx2x3.
     """
-    names = ['xy', 'scale', 'ori']
-    for var_name, var, req_shape in zip(names, [xy, scale, ori], [("B", "N", 2), ("B", "N", 1, 1), ("B", "N", 1)]):
-        if not isinstance(var, torch.Tensor):
-            raise TypeError(f"{var_name} type is not a torch.Tensor. Got {type(var)}")
-        if len(var.shape) != len(req_shape):  # type: ignore  # because it does not like len(tensor.shape)
-            raise TypeError("{} shape should be must be [{}]. " "Got {}".format(var_name, str(req_shape), var.size()))
-        for i, dim in enumerate(req_shape):  # type: ignore # because it wants typing for dim
-            if dim is not int:
-                continue
-            if var.size(i) != dim:
-                raise TypeError(
-                    "{} shape should be must be [{}]. " "Got {}".format(var_name, str(req_shape), var.size())
-                )
```

Comment:
Other alternative apis could be `KORNIA_CHECK_EQ(len(xy.shape), 4)a
Or a more advanced one `KORNIA_CHECK_SHAPE(xy, ("B", "N", 2), "Expected a BxNx2 tensor")

the last can check the following 
1. Is a tensor, so it has a shape
2. (xy.shape) == len(expected shape)
3. For non string positions, checks the actual value `xy.shape[2] == 2`

Reply:
I see. I have added KORNIA_CHECK_SHAPE, but you cannot you non-string, instead use like - all strings, but some contain letters, other - numbers. Also - list, not tuple. All of this because of jit. 

I have also replaced old shape checks with new in kornia.feature, but don't have time for the rest of the library.
```python
KORNIA_CHECK_SHAPE(img, ["B", "1", "H", "W"])
```